source/controller/Award.ts

```diff
@@ -0,0 +1,135 @@
+import {
+    Authorized,
+    Body,
+    CurrentUser,
+    Delete,
+    Get,
+    HttpCode,
+    JsonController,
+    NotFoundError,
+    OnNull,
+    OnUndefined,
+    Param,
+    Patch,
+    Post
+} from 'routing-controllers';
+import { ResponseSchema } from 'routing-controllers-openapi';
+
+import { dataSource, Hackathon, User, Award, Media } from '../model';
+
+import { HackathonController } from './Hackathon';
+import { ActivityLogController } from './ActivityLog';
+
+@JsonController('/hackathon/:hackathonName/award')
+export class AwardController {
+    store = dataSource.getRepository(Award);
+    hackathonStore = dataSource.getRepository(Hackathon);
+
+    @Post()
+    @Authorized()
+    @HttpCode(201)
+    @ResponseSchema(Award)
+    async createOne(
+        @CurrentUser() currentUser: User,
+        @Param('hackathonName') hackathonName: string,
+        @Body()
+        awardData: {
+            name: string;
+            description: string;
+            quantity: number;
+            target: 'team' | 'individual';
+            pictures: Media[];
+        }
+    ) {
+        const hackathon = await this.hackathonStore.findOneBy({
+            name: hackathonName
+        });
+        if (!hackathon) throw new NotFoundError('Hackathon not found');
+
+        await HackathonController.ensureAdmin(currentUser.id, hackathonName);
+
+        const award = this.store.create({
+            ...awardData,
+            hackathon
+        });
+
+        const saved = await this.store.save(award);
+        await ActivityLogController.logCreate(currentUser, 'Award', saved.id);
+        return saved;
+    }
+
+    @Get('/:awardId')
+    @OnNull(404)
+    @ResponseSchema(Award)
+    async getOne(
+        @Param('hackathonName') hackathonName: string,
+        @Param('awardId') awardId: number
+    ) {
+        const award = await this.store
+            .createQueryBuilder('award')
+            .innerJoinAndSelect('award.hackathon', 'hackathon')
+            .where('hackathon.name = :hackathonName', { hackathonName })
+            .andWhere('award.id = :awardId', { awardId })
+            .getOne();
+
+        if (!award) {
+            return null;
+        }
+
+        return award;
+    }
+
+    @Patch('/:awardId')
+    @Authorized()
+    @ResponseSchema(Award)
+    async updateOne(
+        @CurrentUser() currentUser: User,
+        @Param('hackathonName') hackathonName: string,
+        @Param('awardId') awardId: number,
+        @Body()
+        updateData: Partial<
+            Pick<
+                Award,
+                'name' | 'description' | 'quantity' | 'target' | 'pictures'
+            >
+        >
```

Comment:
```suggestion
    @Put('/:awardId')
    @Authorized()
    @ResponseSchema(Award)
    async updateOne(
        @CurrentUser() currentUser: User,
        @Param('hackathonName') hackathonName: string,
        @Param('awardId') awardId: number,
        @Body()
        updateData: Award
```
为了复用创建时的 model，我们现在统一用**全量更新**来实现。